src/System.Runtime/tests/System/LazyTests.cs

```diff
@@ -103,17 +106,283 @@ public static void ToString_DoesntForceAllocation()
+            }
+        }
+
+        private static void DifferentException<T>(Lazy<T> x)
+        {
+            Exception first = null;
+            try
+            {
+                var _ = x.Value;
+            }
+            catch (Exception thrown1)
+            {
+                first = thrown1;
+            }
+            Assert.NotNull(first);
+
+            Exception second = null;
+            try
+            {
+                var _ = x.Value;
+            }
+            catch (Exception thrown2)
+            {
+                second = thrown2;
+            }
+            Assert.NotNull(second);
+
+            Assert.NotEqual(first, second);
+        }
+
+        [Fact]
+        public static void SameOrDifferentException()
+        {
+            SameException(new Lazy<int>(() => { throw new MyException(99); }));
+            SameException(new Lazy<int>(() => { throw new MyException(99); }, true));
+            SameException(new Lazy<int>(() => { throw new MyException(99); }, false));
+            SameException(new Lazy<int>(() => { throw new MyException(99); }, LazyThreadSafetyMode.ExecutionAndPublication));
+            SameException(new Lazy<int>(() => { throw new MyException(99); }, LazyThreadSafetyMode.None));
+
+            DifferentException(new Lazy<int>(() => { throw new MyException(99); }, LazyThreadSafetyMode.PublicationOnly));
+
+            DifferentException(new Lazy<ExceptionInCtor>());
+            DifferentException(new Lazy<ExceptionInCtor>(true));
+            DifferentException(new Lazy<ExceptionInCtor>(false));
+            DifferentException(new Lazy<ExceptionInCtor>(LazyThreadSafetyMode.ExecutionAndPublication));
+            DifferentException(new Lazy<ExceptionInCtor>(LazyThreadSafetyMode.None));
+
+            DifferentException(new Lazy<ExceptionInCtor>(LazyThreadSafetyMode.PublicationOnly));
+
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99)));
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), true));
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), false));
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), LazyThreadSafetyMode.ExecutionAndPublication));
+            SameException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), LazyThreadSafetyMode.None));
+
+            DifferentException(new Lazy<ExceptionInCtor>(() => new ExceptionInCtor(99), LazyThreadSafetyMode.PublicationOnly));
+        }
+
+        [Fact]
+        static void Serialization_ValueType()
+        {
+            var stream = new MemoryStream();
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(stream, new Lazy<int>(() => 42));
+            stream.Seek(0, SeekOrigin.Begin);
+
+            var fortytwo = (Lazy<int>)formatter.Deserialize(stream);
+            Assert.True(fortytwo.IsValueCreated);
+            Assert.Equal(fortytwo.Value, 42);
+        }
```

Comment:
We have a small helper for this in:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Runtime/Serialization/Formatters/BinaryFormatterHelpers.cs
With that this would become something like:
```C#
[Fact]
public static void Serialization_Int32_Roundtrips()
{
    Lazy<int> clone = BinaryFormatterHelpers.Clone(new Lazy<int>(() => 42));
    Assert.True(clone.IsValueCreated);
    Assert.Equal(42, clone.Value);
}
```